### PR TITLE
Feature/maintenance mode

### PR DESCRIPTION
**Pull request by paul-burnett**
Maintenance page that can be shown during deployments/downtime.  
To configure, can set variable MAINTENANCE_MODE_BLOCK_ALL=true (to redirect all pages/users to maintenance) or MAINTENANCE_MODE_BLOCK_NEW=true (to just redirect new users and allow users part way through the journey to continue).

### CODE REVIEW

File: public/static/locales/cy/common.json [hunk 1]
```diff
@@ -382,7 +382,7 @@
   "Accepted": "Derbyniwyd",
   "You have entered an invalid number of partners or a duplicate partner name. Please define between 2-5 partners, using initials or middle name to ensure that each entry is unique.": "Rydych chi wedi nodi nifer annilys o bartneriaid neu enw partner dyblyg. Diffiniwch rhwng 2 a 5 partner, gan ddefnyddio llythrennau cyntaf eu henwau neu enw canol i sicrhau bod pob cofnod yn unigryw.",
   "Please use initials or middle names as necessary to ensure each partner name is unique.": "Defnyddiwch lythrennau cyntaf enwau neu enwau canol i sicrhau bod enw pob partner yn unigryw.",
-
+  "This service is currently undergoing maintenance.  Please try again later.": "This service is currently undergoing maintenance.  Please try again later.",
```
Comment (jj-perry) [inline public/static/locales/cy/common.json:385]: Shouldn't this be the welsh value?

**Review by jj-perry — COMMENTED** [review]
At what point does changing the env variables take effect? Is it just a matter of changing the variables and restarting the app service?

**Comment by paul-burnett** [comment]
> At what point does changing the env variables take effect? Is it just a matter of changing the variables and restarting the app service?

Correct

File: public/static/locales/cy/common.json (hunk 1 above)
Comment (paul-burnett) [reply public/static/locales/cy/common.json:385]: Thanks.  Will update this with translation from google.  Will be sending all the new welsh for this release to FSA for confirmation shortly.

**Review by jj-perry — APPROVED** [review]

**Pull request merged**